packages/python3.rb

```diff
@@ -47,7 +47,14 @@ class Python3 < Package
   conflicts_ok
 
   def self.preinstall
-    system 'crew remove py3_setuptools py3_pip', exception: false
+    @device = JSON.load_file("#{CREW_CONFIG_PATH}/device.json", symbolize_names: true)
+    @replaces = %w[py3_pip py3_setuptools]
+    @replaces.each do |package|
+      if @device[:installed_packages].any? { |elem| elem[:name] == package }
+        puts "Removing superseded package #{package}...".orange
+        system "crew remove #{package}", exception: false
```

Comment:
Does this work?  It would be recursive crew commands.

Reply:
It works!

Reply:
Then you don't need to loop over each package.  This would work too:
```
system "crew remove pip3_pip py3_setuptools", exception: false
```

Reply:
So does this really work or are there issues with nesting crew commands?

Reply:
The nested solution now works with a `crew` change to reload `device.json` after preinstall finishes.

Reply:
Also modified the code to run `crew remove` only once.